File: app/src/main/java/co/fddittmar/planbuilder/contracts/NewProgramContract.java
package co.fddittmar.planbuilder.contracts;

import java.util.List;

import co.fddittmar.planbuilder.data.model.Exercise;

/**
 * Contract for the New Program activity. It has an interface for the View and Presenter.
 */

public class NewProgramContract {
    public interface View {
        void addExercise();
        void saveProgram();
        void returnToMainActivity();
        void setupNewExerciseDialog();
    }

    public interface Actions {

        void saveProgram(String title, String startDate, String endDate, List<Exercise> exercises);
        void onAddNewExercise();
    }
}

File: app/src/main/java/co/fddittmar/planbuilder/contracts/NewProgramContract.kt
package co.fddittmar.planbuilder.contracts

import co.fddittmar.planbuilder.data.model.Exercise

/**
 * Contract for the New Program activity. It has an interface for the View and Presenter.
 */

class NewProgramContract {
    interface View {
        fun addExercise()
        fun saveProgram()
        fun returnToMainActivity()
        fun setupNewExerciseDialog()
    }

    interface Actions {

        fun saveProgram(title: String, startDate: String, endDate: String, exercises: List<Exercise>)
        fun onAddNewExercise()
    }
}
